### PR TITLE
fix some warnings

### PR DESCRIPTION
**Pull request by Megamouse**

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_spu.h [hunk 1]
```diff
@@ -316,7 +316,7 @@ struct lv2_spu_group
 		, run_state(SPU_THREAD_GROUP_STATUS_NOT_INITIALIZED)
 		, exit_status(0)
 		, join_state(0)
-		// TODO: args()
+		, args({}) // TODO
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_spu.h:319]: I hardly remember what this TODO is for, it can be removed.

**Review by RipleyTom — APPROVED** [review]
Only reviewed what was of interest to me(np and sys_net stuff).

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 2]
```diff
@@ -1161,7 +1161,7 @@ static void ppu_check_patch_spu_images(const ppu_module& mod, const ppu_segment&
 
 					// FIXME: This seems to terminate SPU code prematurely in some cases
 					// Likely due to absolute branches
-					if (spu_thread::is_exec_code(instruction, {reinterpret_cast<const u8*>(ls_segment.data()), ls_segment.size()}, 0))
+					if (spu_thread::is_exec_code(::narrow<u32>(instruction), {reinterpret_cast<const u8*>(ls_segment.data()), ls_segment.size()}, 0))
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.cpp:1164]: make it u32 instruction; instead with static cast

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 3]
```diff
@@ -1148,7 +1148,7 @@ static void ppu_check_patch_spu_images(const ppu_module& mod, const ppu_segment&
 
 			for (usz addr_last = 0, valid_count = 0, invalid_count = 0;;)
 			{
-				usz instruction = ls_segment.find("\x24\0\x40\x80"sv, addr_last);
+				const usz instruction = ls_segment.find("\x24\0\x40\x80"sv, addr_last);
```
Comment (elad335) [inline rpcs3/Emu/Cell/PPUModule.cpp:1151]: u32 here

File: rpcs3/Emu/Cell/PPUModule.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/PPUModule.cpp]: like this? idk if the find is guaranteed to be in scope of u32

Comment (elad335) [reply rpcs3/Emu/Cell/PPUModule.cpp]: yeah it is 👍

Comment (Megamouse) [reply rpcs3/Emu/Cell/PPUModule.cpp]: actually, won't this break the umax check?

Comment (elad335) [reply rpcs3/Emu/Cell/PPUModule.cpp]: Suprisingly no

Comment (Megamouse) [reply rpcs3/Emu/Cell/PPUModule.cpp]: Ah, because casting from 0xFFFFFFFFFFFFFFFF to u32 is 0xFFFFFFFF.

Comment (Megamouse) [reply rpcs3/Emu/Cell/PPUModule.cpp]: Anything else?

File: rpcs3/Emu/NP/rpcn_client.cpp [hunk 4]
```diff
@@ -759,7 +759,14 @@ namespace rpcn
 			addr_rpcn.sin_port   = std::bit_cast<u16, be_t<u16>>(port); // htons
 			addr_rpcn.sin_family = AF_INET;
 
+#ifdef WIN32
```
Comment (oltolm) [inline rpcs3/Emu/NP/rpcn_client.cpp:762]:
```
#ifdef _MSC_VER
```

File: rpcs3/Emu/NP/rpcn_client.cpp [hunk 5]
```diff
@@ -759,7 +759,14 @@ namespace rpcn
...
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#endif
 			hostent* host_addr = gethostbyname(splithost[0].c_str());
+#ifdef WIN32
```
Comment (oltolm) [inline rpcs3/Emu/NP/rpcn_client.cpp:767]:
```
#ifdef _MSC_VER
```

**Pull request merged**